region/maridia/inner-green/Shaktool Room.json

```
@@ -96,6 +126,72 @@
       "flashSuitChecked": true,
       "devNote": "Entering the room with the flag 'f_ShaktoolDoneDigging' set means that the sand blocks and Power Bomb blocks will not spawn."
     },
+    {
+      "link": [1, 1],
+      "name": "R-Mode Spark Interrupt (Gain Blue Suit)",
+      "requires": [
+        {"obstaclesCleared": ["R-Mode", "B"]},
+        {"or": [
+          "h_CrystalFlashForReserveEnergy",
+          {"and": [
+            "h_RModeCanRefillReserves",
+            {"or": [
+              {"and": [
+                {"obstaclesCleared": ["A"]},
+                {"or": [
+                  {"and": [
+                    "canWaterShineCharge",
+                    {"canShineCharge": {"usedTiles": 35, "openEnd": 2}}
+                  ]},
+                  {"and": [
+                    "Gravity",
+                    {"canShineCharge": {"usedTiles": 42, "openEnd": 2}}
+                  ]}
+                ]},
```

Long runways are mostly only needed for cross room strats and maybe very low-level strats.
```suggestion
                "h_shinechargeMaxRunway",
```